### DISCUSSION TITLE
Cookie consent form (privacy laws)

### DISCUSSION THREAD

**Discussion by rickracs** [adityatelange/hugo-PaperMod#825] [Q&A]
Is it possible to add a cookie consent form for Google analytics? I tried various online resources on how to create one, but I think I'm not knowledgeable enough to get it working.

**Comment by lorenviol** [comment]
Try looking into onetrust.com , they offer one for free, I implemented it into my hugo website

**Comment by cloudfright** [comment]
See also https://littlebigtech.net/posts/hugo-gdpr-cookie-consent-banner/

**Reply by cloudfright** [reply]
[Here's how](https://blog.cloudfright.com/posts/hello-hugo-pt3/) I tackled it for App Insights (using a variation on the link above).

**Reply by frascu** [reply]
Hi @cloudfright @kudos01,
I created a forked [repository](https://github.com/frascu/hugo-PaperMod) with your changes to make a pull request and add your feature to the Hugo template. 

My idea is to add a property called `cookieEnabled` that can enable this banner directly in the configuration.

Given that you share the code, I want your authorization to do this.
What do you think about it? Do you agree?

**Reply by Kudos01** [reply]
Yes go ahead!

**Reply by danielwagn3r** [reply]
@cloudfright Where you able to bring the above mentioned approach to work with PaperMod? I'd really love the idea of having this as a feature in the theme - it's an essential functionality for each website which uses Cookies and/or Third-Party dependencies like Analytics, Fonts or so.

I tried using https://hugocodex.org/add-ons/cookie-consent/ but didn't get it working properly.

**Reply by danielwagn3r** [reply]
Hi again, I played around a little bit and modified extend_head.html a little bit, see https://gist.github.com/danielwagn3r/1c6627f7ec2ba0c229d8b223e60694f0 , to use the standard configuration option `GoogleAnalytics ` from `config.yaml`

I saw that a cookie `_ga_XXX` is set wether I click yes or now in the banner, is this intended?

**Reply by Kudos01** [reply]
I just double checked and you are right, I must have changes something that introduced this bug. This is not intended behavior.

I will look at a fix as soon as I can.

**Reply by Kudos01** [reply]
Hello again.
I took a look and managed to fix it. The changes are on my repo:

https://github.com/Kudos01/Hugo-GDPR-Cookie-Consent-Banner

The delete cookie function was not working properly, since the domain was not specified. Also the SameSite attribute was not specified correctly, so I fixed that too. You can change the SameSite attribute to whatever you want, just follow these guidelines:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite

Now the GA cookies are deleted (or attempted to) in all cases, when the cookie-consent cookie is false and also when it does not exist.

Feel free to suggest improvements! These changes on their own at least get it working properly.

**Reply by geeqla** [reply]
@adityatelange would you be open to including a basic "cookie banner" that theme users could activate through configuration? For example, similar to the one on your website https://adityatelange.in. This is not GDPR-compliant, but at least something.
